### ISSUE TITLE
Show/Hide Point Primitive

### ISSUE THREAD

**Issue by QuantumMatter** [darwin-education/resium#101]
Hello everyone, thanks for taking the time to read this. I'm trying to load a couple thousand PointPrimitive objects on the globe, and then control whether or not they are visible based on the state. I've tried using the show prop of the PointPrimitive, but changing it after the object has already been rendered seems to have no effect. Doing the same thing with an Entity works, but unfortunately I need to use the PointPrimitive because of the massive amount of data I'm trying to render. I've included a simple example that demonstrates the issue; any help would be greatly appreciated.

```javascript
import React from 'react';

import { Viewer, PointPrimitive, PointPrimitiveCollection } from "resium";
import { Cartesian3 } from 'cesium';
const pointPosition = Cartesian3.fromDegrees(-105.2177025, 40.021614, 1700);

class App extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            show: true
        }
    }

    componentDidMount() {

        setTimeout(() => {
            console.log("Trying to hide point");
            this.setState({
                show: !this.state.show
            });
        }, 5000);
    }

    render() {
        return (
            <Viewer>
                <PointPrimitiveCollection>
                    <PointPrimitive position={ pointPosition } show={ this.state.show } />
                </PointPrimitiveCollection>
            </Viewer>
        );
    }
}

export default App;
```

**Comment by QuantumMatter** [comment]
Follow up: using the Cesium library without the react wrapper seems to work fine. Here's the code I used to test that

```javascript
import React from 'react';

import { Viewer } from "resium";
import Cesium from 'cesium';

class App extends React.Component {

    constructor(props) {
        super(props);

        this.point = React.createRef();
        this.viewer = null; React.createRef();

        this.state = {
            show: true
        }
    }

    componentDidMount() {
        this.points = this.viewer.scene.primitives.add(new Cesium.PointPrimitiveCollection());
        this.points.add({
            position: Cesium.Cartesian3.fromDegrees(-105.2177025, 40.021614, 1700),
            color: Cesium.Color.YELLOW,
            show: true
        });
        
        setTimeout(() => {
            console.log("Trying to hide point manually: ", this.points.get(0));
            this.points.get(0).show = false;
        }, 5000);
    }

    render() {
        return (
            <Viewer ref={ ref => { this.viewer = ref.cesiumElement } } />
        );
    }
}

export default App;
```

**Comment by QuantumMatter** [comment]
I was able to track down the issue to [when the PointPrimitive added to the collection](https://github.com/darwin-education/resium/blob/master/src/PointPrimitive/PointPrimitive.ts#L71). For some reason, calling
```javascript
PointPrimitiveCollection#add
```
returns a new PointPrimitive object with all of the same attributes as the one passed as the argument. Therefore, when the resium library tries to update the component, it's trying to update the cached component that was initialized in the 'create' method and not the one existing in the PointPrimitiveCollection. I made a quick and dirty Custom Point Primitive class that demonstrates how to fix the issue. I'd make a pull request, but I'm not entirely sure how to do that, or work the fix back into the library directly.

``` javascript
import React from 'react';

import { withCesium } from "resium";
import { PointPrimitive } from "cesium";

import pick from "object.pick";
import shadows from '@material-ui/core/styles/shadows';

class CustomPointPrimitive extends React.Component {

    constructor(props) {
        super(props);

        this.cesiumElement = new PointPrimitive(this.getCesiumProps(props));
    }

    cesiumProps = [
        "color",
        "disableDepthTestDistance",
        "distanceDisplayCondition",
        "id",
        "outlineColor",
        "outlineWidth",
        "pixelSize",
        "position",
        "scaleByDistance",
        "show",
        "translucencyByDistance",
    ];

    shouldUpdate(prevProps, props) {
        console.log("[CustomPointPrimitive]::shouldUpdate", prevProps, props);
        const dirty = Object.keys(prevProps).filter(key => prevProps[key] != props[key]);
        dirty.forEach(key => {
            console.log("[CustomPointPrimitive]::shouldUpdate DIRTY " + key, prevProps[key], props[key]);
        })
        return dirty;
    }

    getCesiumProps(props) {
        return pick(props, this.cesiumProps);
    }

    componentDidMount() {
        console.log("[CustomPointPrimitive]::didMount()");

        if (this.props.cesium.pointPrimitiveCollection) {
            console.log("[CustomPointPrimitive]::didMount() Proper Context!");
            const newElement = this.props.cesium.pointPrimitiveCollection.add(this.cesiumElement);
            console.log("[CustomPointPrimitive]::didMount() Comparing newElement to old element", this.cesiumElement, newElement, this.cesiumElement === newElement);
            this.cesiumElement = newElement;
        }
    }

    componentDidUpdate(prevProps) {
        // Check if mounted
        // Check if read-only props changed

        const needsUpdate = this.shouldUpdate(this.getCesiumProps(prevProps), this.getCesiumProps(this.props));
        console.log("[CustomPointPrimitive]::didUpdate needsUpdate", needsUpdate);
        needsUpdate.forEach(attr => {
            this.cesiumElement[attr] = this.props[attr];
        })
    }

    componentWillUnmount() {
        console.log("[CustomPointPrimitive]::willUnmount()");
    }

    render() {
        console.log("[CustomPointPrimitive]::render()");
        return null;
    }

}

export default withCesium(CustomPointPrimitive);
```

**Comment by rot1024** [comment]
Sorry for the late reply. It's a bug.

I have fixed the bug and released v1.6.0. Please try it!

Thank you.

**Comment by QuantumMatter** [comment]
That fixed my issue! Thanks for the help!
